src/main/java/com/bmuschko/gradle/docker/tasks/image/DockerPushImage.java

```diff
@@ -80,14 +83,30 @@ public void registryCredentials(Action<? super DockerRegistryCredentials> action
     }
 
     private ResultCallback.Adapter<PushResponseItem> createCallback(final Action nextHandler) {
+        // Workaround to manually handle error logic - see https://github.com/docker-java/docker-java/issues/2140
         return new ResultCallback.Adapter<PushResponseItem>() {
+            @Nullable
+            private PushResponseItem latestItem = null;
+
             @Override
             public void onNext(PushResponseItem item) {
+                this.latestItem = item;
+
                 if (nextHandler != null) {
                     nextHandler.execute(item);
                 }
             }
 
+            @Override
+            protected void throwFirstError() {
+                super.throwFirstError();
+
+                if (latestItem == null) {
+                    throw new DockerClientException("Could not push image");
```

Comment:
Is there a way to get the root cause of this failure e.g. one that indicates that incorrect credentials were used?